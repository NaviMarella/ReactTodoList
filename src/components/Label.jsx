import React from "react";
import PropTypes from 'prop-types';
import "./label.css";
import Conditional from "./Conditional";

const Label = ({text, children, hideText, crossText, ...otherProps}) => {
    var styles = [
        "label_text",
        (crossText) ? "label_crossed" : ""
    ].join(" ");

    return(
        <div className="label" {...otherProps}>
            {children}
            <Conditional condition={!hideText}>
                <span className={styles}>{text}</span>
            </Conditional>
        </div>
    )
}

Label.propTypes = {
    text: PropTypes.string.isRequired,
    hideText: PropTypes.bool,
    crossText: PropTypes.bool
};


export default Label