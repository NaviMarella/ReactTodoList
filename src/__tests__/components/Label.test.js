import { shallow } from 'enzyme';
import React from 'react';
import Label from './../../components/Label';

describe('Label Component', () => {
  test('render snapshot', () => {
      let props = {
        text: "Label1",
        hideText: false,
        crossText: false,
      }
    const wrapper = shallow(<Label {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render snapshot with hideText', () => {
      let props = {
        text: "Label1",
        hideText: true,
        crossText: false,
      }
    const wrapper = shallow(<Label {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render snapshot with crossText', () => {
      let props = {
        text: "Label1",
        hideText: false,
        crossText: true,
      }
    const wrapper = shallow(<Label {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
