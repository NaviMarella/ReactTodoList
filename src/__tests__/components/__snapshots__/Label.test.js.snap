// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    crossText={false}
    hideText={false}
    text="Label1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <Conditional
          condition={true}
        >
          <span
            className="label_text "
          >
            Label1
          </span>
        </Conditional>,
      ],
      "className": "label",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span
            className="label_text "
          >
            Label1
          </span>,
          "condition": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Label1",
            "className": "label_text ",
          },
          "ref": null,
          "rendered": "Label1",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <Conditional
            condition={true}
          >
            <span
              className="label_text "
            >
              Label1
            </span>
          </Conditional>,
        ],
        "className": "label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="label_text "
            >
              Label1
            </span>,
            "condition": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Label1",
              "className": "label_text ",
            },
            "ref": null,
            "rendered": "Label1",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Label Component render snapshot with crossText 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    crossText={true}
    hideText={false}
    text="Label1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <Conditional
          condition={true}
        >
          <span
            className="label_text label_crossed"
          >
            Label1
          </span>
        </Conditional>,
      ],
      "className": "label",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span
            className="label_text label_crossed"
          >
            Label1
          </span>,
          "condition": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Label1",
            "className": "label_text label_crossed",
          },
          "ref": null,
          "rendered": "Label1",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <Conditional
            condition={true}
          >
            <span
              className="label_text label_crossed"
            >
              Label1
            </span>
          </Conditional>,
        ],
        "className": "label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="label_text label_crossed"
            >
              Label1
            </span>,
            "condition": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Label1",
              "className": "label_text label_crossed",
            },
            "ref": null,
            "rendered": "Label1",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Label Component render snapshot with hideText 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    crossText={false}
    hideText={true}
    text="Label1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <Conditional
          condition={false}
        >
          <span
            className="label_text "
          >
            Label1
          </span>
        </Conditional>,
      ],
      "className": "label",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span
            className="label_text "
          >
            Label1
          </span>,
          "condition": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Label1",
            "className": "label_text ",
          },
          "ref": null,
          "rendered": "Label1",
          "type": "span",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <Conditional
            condition={false}
          >
            <span
              className="label_text "
            >
              Label1
            </span>
          </Conditional>,
        ],
        "className": "label",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span
              className="label_text "
            >
              Label1
            </span>,
            "condition": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Label1",
              "className": "label_text ",
            },
            "ref": null,
            "rendered": "Label1",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
